### ISSUE TITLE
Float and Double values from request appends unwanted numbers.

### ISSUE THREAD

**Issue by abhayjatindoshi** [laravel/framework#24069]
- Laravel Version: 5.6.18
- PHP Version: 7.1.14

### Description:
float and double numbers when sent in a HTTP request it appends a lot of numbers on the mantissa part. 

### Steps To Reproduce:
Sending the following request
**POST** /api/wallet
```json
{
    "name":"Personal wallet",
    "initial_amount":123.45
}
```

Log::debug($request->all())
```
[2018-05-01 15:55:46] local.DEBUG: array (
  'name' => 'Personal wallet',
  'initial_amount' => 123.4500000000000028421709430404007434844970703125,
)
```

**Route definition:**
```
Route::post("wallet",function(Request $request){
    Log::debug($request->all());
});
```

**Comment by sisve** [comment]
This looks related to json (de)serialization. What you're seeing is the actual correct values, while `123.45` is technically wrong, but a simplified display for human eyes. There's a lot of information about this at [The PHP floating point precision is wrong by default](https://www.leaseweb.com/labs/2013/06/the-php-floating-point-precision-is-wrong-by-default/), and I would urge everyone to correct the precision setting.

Anyhow, there are two precision settings. One (precision) is for humans and is used when a floating point number is converted to string in the normal ways. Another (serialize_precision) is used when json serializes values, which happens by the logging system. This value is higher to make sure you're not losing data when serializing and deserializing a value.

The RFC where this was implemented in PHP 7.1 can be  found at [rfc:precise_float_value](https://wiki.php.net/rfc/precise_float_value).

This is not a Laravel issue, but expected (or at least intended) behavior. You can set your serialize_precision to a lower value and risk losing data when serializing.

**Comment by sisve** [comment]
Yes, I am using 17 in my applications, and I believe the world would be a better place if everyone used it. Mostly because I think it's important to know how floating point numbers work, but also because I like to spread pain and suffering around me.

If you're like me, preferring agony and anguish; https://randomascii.wordpress.com/category/floating-point/ has some some really good articles related to floating point numbers.
